src/maps/runGraphData.ts

```diff
@@ -41,11 +50,16 @@ export const mapRunGraphDataResponse: MapFunction<RunGraphDataResponse, RunGraph
     return this.map('RunGraphArtifactResponse', artifact, 'RunGraphArtifact')
   }) ?? []
 
+  const states: RunGraphStateEvent[] = source.states?.map(state => {
+    return this.map('RunGraphStateResponse', state, 'RunGraphStateEvent')
+  }) ?? []
```

Comment:
Can't you just pass the array of states into the mapper and it'll return an array or undefined?